droopler.install

```diff
@@ -1378,3 +1380,60 @@ function droopler_update_8115() {
   $updater->checklist()
     ->saveProgress(['v2.2' => $values]);
 }
+
+/**
+ * Droopler 2.2 Migrate column count field.
+ */
+function droopler_update_8116(&$sandbox) {
+  if (!isset($sandbox['total'])) {
+    $query = \Drupal::entityQuery('paragraph')
+      ->exists('field_d_number')
+      ->execute();
+
+    $sandbox['total'] = count($query);
+    $sandbox['current'] = 0;
+  }
+
+  $paragraphs_per_batch = 25;
+
+  $query = \Drupal::entityQuery('paragraph')
+    ->exists('field_d_number')
+    ->range($sandbox['current'], $sandbox['current'] + $paragraphs_per_batch)
+    ->execute();
+
+  foreach ($query as $pid) {
+    $paragraph = \Drupal::entityTypeManager()
+      ->getStorage('paragraph')
+      ->load($pid);
+    $settings = $paragraph->get('field_d_settings') ? $paragraph->get('field_d_settings')->value : "{}";
+
+    $json = json_decode($settings);
+    if (empty($json->column_count)) {
+      $json->column_count = $paragraph->field_d_number ? ($paragraph->field_d_number->value ?? '4') : '4';
+      $paragraph->set('field_d_settings', json_encode($json));
+      $paragraph->save();
+    }
+    $paragraph->save();
+    $sandbox['current']++;
+  }
+
+  \Drupal::messenger()->addMessage($sandbox['current'] . ' paragraphs processed.');
+  $sandbox['#finished'] = $sandbox['total'] == 0 ? 1 : $sandbox['current'] / $sandbox['total'];
+
+  if ($sandbox['#finished'] === 1) {
+    $bundles = [
+      'd_p_carousel',
+      'd_p_group_of_counters',
+      'd_p_group_of_text_blocks',
+    ];
+    FieldStorageConfig::loadByName('paragraph', 'field_d_number')->delete();
+
+    foreach ($bundles as $bundle) {
+      $field = FieldConfig::loadByName('paragraph', $bundle, 'field_d_number');
+
+      if ($field !== NULL) {
+        $field->delete();
+      }
+    }
+  }
```

Comment:
jak juz zakonczy, to dodaj jakies info o tym kroku updateu:
 $values['droopler__droopler_update_8116_1'] = TRUE

no i cos tu:
![image](https://user-images.githubusercontent.com/37533838/87317798-c6ec6900-c527-11ea-8274-e0f9b3e80320.png)

Reply:
to dodam ofc, jak update cfg będzie śmigał

Reply:
no tak, ale to wtedy mozesz tez dodac 816_2 - config import  :thinking: 
tylko pytanie czy to potrzebne bedzie w tym wypadku. Tutaj tak czy siak zapiszemy takie info, zeby user widzial, ze przenieslismy wartosci no i zeby widzial status tego przenoszenia, cos moze nie zaskoczyc i nie bedzie działało